app.py

```diff
@@ -315,11 +325,18 @@ def parse():
             else:
                 error_template = hedy_errors[E.error_code]
                 response["Error"] = error_template.format(**E.arguments)
+            if lang in supported_lang:
+                response.update(gradual_feedback_model(code, level, gradual_feedback, lang, E, hedy_exception=True))
         except Exception as E:
             traceback.print_exc()
             print(f"error transpiling {code}")
             response["Error"] = str(E)
-    logger.log ({
+            if lang in supported_lang:
+                response.update(gradual_feedback_model(code, level, gradual_feedback, lang, E, hedy_exception=False))
+        if lang in supported_lang:
+            session['code'] = code
```

Comment:
Same here, we should use the set_session_var function.

Reply:
Fixed!